### ISSUE TITLE
no-shadowed-variable breaks with exports from namespaces

### ISSUE THREAD

**Issue by domoritz** [palantir/tslint#3214]
### Bug Report

- __TSLint version__: 5.7
- __TypeScript version__: 2.5
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
export namespace Foo {
  export const ANSWER = 42;
}
export const ANSWER = Foo.ANSWER;
```

with `tslint.json` configuration:

```json
"no-shadowed-variable": true
```

#### Actual behavior

Getting

```ts
[tslint] Shadowed name: 'ANSWER' (no-shadowed-variable)
const Foo.ANSWER: 42
```

#### Expected behavior

No error.

**Comment by domoritz** [comment]
@ajafff's options for disabling the check for some cases also doesn't seem to cover this case. I cannot silence the error from above.

**Comment by freebroccolo** [comment]
I'm also running into this now. Is this a regression?

**Comment by domoritz** [comment]
I'm stuck with an ancient version of tslint because of this.

**Comment by ajafff** [comment]
This is basically a duplicate of #3078. The option requested there will also fix this issue. I'll submit a PR shortly.

**Comment by domoritz** [comment]
The fix requires you to use `temporalDeadZone`.
